addons/sale/models/sale.py

```diff
@@ -357,13 +357,13 @@ def onchange_partner_id(self):
             values['note'] = self.with_context(lang=self.partner_id.lang).env.company.invoice_terms
 
         # Use team of salesman if any otherwise leave as-is
-        values['team_id'] = partner_user.team_id.id if partner_user and partner_user.team_id else self.team_id
+        values['team_id'] = self.env['crm.team']._get_default_team_id(user_id=user_id) or self.team_id
```

Comment:
`_get_default_team_id(user_id) if partner_user else self.team_id` instead,
we will never fall back on the `or self.team_id` with your current change because `_get_default_team_id` should always return something in this case.